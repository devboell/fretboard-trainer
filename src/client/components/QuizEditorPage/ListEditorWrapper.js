import styled from 'styled-components'

const ListEditorWrapper = styled.div`
  display: flex;
  height: 300px;
  padding: 20px;
  border-radius: 10px;
  border: 2px solid gray;
`

export default ListEditorWrapper
