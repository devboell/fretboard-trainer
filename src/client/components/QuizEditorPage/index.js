import React from 'react'
import pt from 'prop-types'
import { isEmpty } from 'ramda'

import List from 'components/List'
import QuizEditor from 'components/QuizEditor'
import Wrapper from './Wrapper'

const newQuiz = {
  id: undefined,
  name: '',
  __typename: 'Quiz',
}

const QuizEditorPage = ({
  quizzes,
  selectedQuizId,
  isNew,
  createQuiz,
  updateQuiz,
  deleteQuiz,
  selectQuiz,
  clearQuizSelection,
}) => {
  const selectedQuiz = selectedQuizId === 'no_selection' || isEmpty(quizzes)
    ? undefined
    : quizzes.filter(q => q.id === selectedQuizId)[0]
  const quiz = isNew ? newQuiz : selectedQuiz

  return (
    <Wrapper>
      <List
        items={quizzes}
        selectedItemId={selectedQuizId}
        onSelectItem={selectQuiz}
      />
      <QuizEditor
        {...{
          quiz,
          isNew,
          clearQuizSelection,
        }}
        onCreateQuiz={createQuiz}
        onUpdateQuiz={updateQuiz}
        onDeleteQuiz={deleteQuiz}
      />
    </Wrapper>
  )
}

QuizEditorPage.propTypes = {
  quizzes: pt.arrayOf(pt.shape({})).isRequired,
  selectedQuizId: pt.string.isRequired,
  isNew: pt.bool.isRequired,
  createQuiz: pt.func.isRequired,
  updateQuiz: pt.func.isRequired,
  deleteQuiz: pt.func.isRequired,
  selectQuiz: pt.func.isRequired,
  clearQuizSelection: pt.func.isRequired,
}

export default QuizEditorPage
