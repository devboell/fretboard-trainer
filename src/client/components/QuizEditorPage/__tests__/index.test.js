import React from 'react'
import testData from 'fixtures/quizEditor'
import QuizEditorPage from '../index'


describe('QuizEditorPage, no selection', () => {
  const data = testData('no_selection', false)
  const wrapper = mount(<QuizEditorPage {...data} />)

  it('snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('> List > onSelectItem', () => {
    const button = wrapper.find('List').find('button').at(2)
    button.simulate('click')
    expect(data.selectQuiz).toBeCalledWith('3')
  })

  it('> CreateWrapper > EditorButton', () => {
    const button = wrapper.find('CreateWrapper').find('EditorButton')
    button.simulate('click')
    expect(data.clearQuizSelection).toBeCalledWith(true)
  })
})

describe('QuizEditorPage, selection = 1', () => {
  const data = testData('1', false)
  const wrapper = mount(<QuizEditorPage {...data} />)

  it('snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('> QuizForm > button > save(updateQuiz)', () => {
    const inputEvent = {
      target: {
        type: 'text',
        name: 'name',
        value: 'updated',
      },
    }
    const updateQuizArg = {
      id: '1',
      name: 'updated',
      type: 'pc',
      __typename: 'Quiz',
    }

    const quizForm = wrapper.find('QuizForm')
    const input = quizForm.find('input[name="name"]')
    const saveButton = quizForm.find('button').at(0)

    input.simulate('change', inputEvent)
    saveButton.simulate('submit')

    expect(data.updateQuiz).toBeCalledWith(updateQuizArg)
  })

  it('> QuizForm > button > delete', () => {
    const deleteButton = wrapper.find('QuizForm').find('button').at(1)
    deleteButton.simulate('click')
    expect(data.deleteQuiz).toBeCalledWith({ id: '1' })
  })
})

describe('QuizEditorPage, isNew', () => {
  const data = testData('no_selection', true)

  const wrapper = mount(<QuizEditorPage {...data} />)
  jest.clearAllMocks()
  it('snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('> QuizForm > button > save(createQuiz)', () => {
    const inputEvent = {
      target: {
        type: 'text',
        name: 'name',
        value: 'new',
      },
    }
    const createQuizArg = {
      id: undefined,
      name: 'new',
      type: 'pc',
      __typename: 'Quiz',
    }

    const quizForm = wrapper.find('QuizForm')
    const input = quizForm.find('input[name="name"]')
    const saveButton = quizForm.find('button').at(0)

    input.simulate('change', inputEvent)
    saveButton.simulate('submit')

    expect(data.createQuiz).toBeCalledWith(createQuizArg)
  })

  it('> QuizForm > button > delete, should be disabled', () => {
    const deleteButton = wrapper.find('QuizForm').find('button').at(1)
    deleteButton.simulate('click')
    expect(data.deleteQuiz).not.toHaveBeenCalled()
  })
})

