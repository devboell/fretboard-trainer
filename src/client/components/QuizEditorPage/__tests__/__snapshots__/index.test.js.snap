// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizEditorPage, isNew snapshot 1`] = `
<QuizEditorPage
  clearQuizSelection={[MockFunction]}
  createQuiz={[MockFunction]}
  deleteQuiz={[MockFunction]}
  isNew={true}
  quizzes={
    Array [
      Object {
        "__typename": "Quiz",
        "id": "1",
        "name": "quiz one",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "2",
        "name": "quiz two",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "3",
        "name": "quiz three",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "4",
        "name": "quiz four",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "5",
        "name": "quiz five",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "6",
        "name": "quiz six",
        "type": "pc",
      },
    ]
  }
  selectQuiz={[MockFunction]}
  selectedQuizId="no_selection"
  updateQuiz={[MockFunction]}
>
  <Wrapper>
    <div
      className="Wrapper-ThVjl hrxOjE"
    >
      <List
        items={
          Array [
            Object {
              "__typename": "Quiz",
              "id": "1",
              "name": "quiz one",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "2",
              "name": "quiz two",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "3",
              "name": "quiz three",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "4",
              "name": "quiz four",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "5",
              "name": "quiz five",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "6",
              "name": "quiz six",
              "type": "pc",
            },
          ]
        }
        onSelectItem={[MockFunction]}
        selectedItemId="no_selection"
      >
        <Wrapper>
          <div
            className="Wrapper-ThVjl iVApgD"
          >
            <Ul>
              <ul
                className="Ul-bfMUEV cngFat"
              >
                <Li
                  key="1"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz one
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="2"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz two
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="3"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz three
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="4"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz four
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="5"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz five
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="6"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz six
                      </button>
                    </Button>
                  </li>
                </Li>
              </ul>
            </Ul>
          </div>
        </Wrapper>
      </List>
      <QuizEditor
        clearQuizSelection={[MockFunction]}
        isNew={true}
        onCreateQuiz={[MockFunction]}
        onDeleteQuiz={[MockFunction]}
        onUpdateQuiz={[MockFunction]}
        quiz={
          Object {
            "__typename": "Quiz",
            "id": undefined,
            "name": "",
            "type": "pc",
          }
        }
      >
        <Wrapper>
          <div
            className="Wrapper-ThVjl eCjdPG"
          >
            <CreateWrapper>
              <div
                className="CreateWrapper-kzyVjd gyhNsT"
              >
                <EditorButton
                  disabled={true}
                  onClick={[Function]}
                >
                  <button
                    className="EditorButton-chDRFm dVoSaf"
                    disabled={true}
                    onClick={[Function]}
                  >
                    New
                  </button>
                </EditorButton>
              </div>
            </CreateWrapper>
            <FormWrapper>
              <div
                className="FormWrapper-hxeBuG qScKG"
              >
                <QuizForm
                  clearQuizSelection={[MockFunction]}
                  isNew={true}
                  onCreateQuiz={[MockFunction]}
                  onDeleteQuiz={[MockFunction]}
                  onUpdateQuiz={[MockFunction]}
                  quiz={
                    Object {
                      "__typename": "Quiz",
                      "id": undefined,
                      "name": "",
                      "type": "pc",
                    }
                  }
                >
                  <Form
                    onSubmit={[Function]}
                  >
                    <form
                      className="Form-dhHGoS fGKlyR"
                      onSubmit={[Function]}
                    >
                      <FieldsWrapper>
                        <div
                          className="FieldsWrapper-eDhoxC fXMMNv"
                        >
                          <Label
                            htmlFor="name"
                          >
                            <label
                              className="Label-bxMjsr hSpJCw"
                              htmlFor="name"
                            >
                              name:
                              <InputText
                                id="name"
                                name="name"
                                onChange={[Function]}
                                placeholder="Enter a name"
                                type="text"
                                value=""
                              >
                                <input
                                  className="InputText-ektxEQ kaKplR"
                                  id="name"
                                  name="name"
                                  onChange={[Function]}
                                  placeholder="Enter a name"
                                  type="text"
                                  value=""
                                />
                              </InputText>
                            </label>
                          </Label>
                          <Label
                            htmlFor="type"
                          >
                            <label
                              className="Label-bxMjsr hSpJCw"
                              htmlFor="type"
                            >
                              type:
                              <input
                                checked={true}
                                id="type"
                                name="type"
                                onChange={[Function]}
                                type="radio"
                                value="pc"
                              />
                            </label>
                          </Label>
                        </div>
                      </FieldsWrapper>
                      <ButtonsWrapper>
                        <div
                          className="ButtonsWrapper-xBhhP evhcHL"
                        >
                          <EditorButton
                            disabled={true}
                            type="submit"
                          >
                            <button
                              className="EditorButton-chDRFm dVoSaf"
                              disabled={true}
                              type="submit"
                            >
                              Save
                            </button>
                          </EditorButton>
                          <EditorButton
                            disabled={true}
                            onClick={[Function]}
                          >
                            <button
                              className="EditorButton-chDRFm dVoSaf"
                              disabled={true}
                              onClick={[Function]}
                            >
                              Delete
                            </button>
                          </EditorButton>
                        </div>
                      </ButtonsWrapper>
                    </form>
                  </Form>
                </QuizForm>
              </div>
            </FormWrapper>
          </div>
        </Wrapper>
      </QuizEditor>
    </div>
  </Wrapper>
</QuizEditorPage>
`;

exports[`QuizEditorPage, no selection snapshot 1`] = `
<QuizEditorPage
  clearQuizSelection={[MockFunction]}
  createQuiz={[MockFunction]}
  deleteQuiz={[MockFunction]}
  isNew={false}
  quizzes={
    Array [
      Object {
        "__typename": "Quiz",
        "id": "1",
        "name": "quiz one",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "2",
        "name": "quiz two",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "3",
        "name": "quiz three",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "4",
        "name": "quiz four",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "5",
        "name": "quiz five",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "6",
        "name": "quiz six",
        "type": "pc",
      },
    ]
  }
  selectQuiz={[MockFunction]}
  selectedQuizId="no_selection"
  updateQuiz={[MockFunction]}
>
  <Wrapper>
    <div
      className="Wrapper-ThVjl hrxOjE"
    >
      <List
        items={
          Array [
            Object {
              "__typename": "Quiz",
              "id": "1",
              "name": "quiz one",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "2",
              "name": "quiz two",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "3",
              "name": "quiz three",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "4",
              "name": "quiz four",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "5",
              "name": "quiz five",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "6",
              "name": "quiz six",
              "type": "pc",
            },
          ]
        }
        onSelectItem={[MockFunction]}
        selectedItemId="no_selection"
      >
        <Wrapper>
          <div
            className="Wrapper-ThVjl iVApgD"
          >
            <Ul>
              <ul
                className="Ul-bfMUEV cngFat"
              >
                <Li
                  key="1"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz one
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="2"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz two
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="3"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz three
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="4"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz four
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="5"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz five
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="6"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz six
                      </button>
                    </Button>
                  </li>
                </Li>
              </ul>
            </Ul>
          </div>
        </Wrapper>
      </List>
      <QuizEditor
        clearQuizSelection={[MockFunction]}
        isNew={false}
        onCreateQuiz={[MockFunction]}
        onDeleteQuiz={[MockFunction]}
        onUpdateQuiz={[MockFunction]}
      >
        <Wrapper>
          <div
            className="Wrapper-ThVjl eCjdPG"
          >
            <CreateWrapper>
              <div
                className="CreateWrapper-kzyVjd gyhNsT"
              >
                <EditorButton
                  disabled={false}
                  onClick={[Function]}
                >
                  <button
                    className="EditorButton-chDRFm dVoSaf"
                    disabled={false}
                    onClick={[Function]}
                  >
                    New
                  </button>
                </EditorButton>
              </div>
            </CreateWrapper>
            <FormWrapper>
              <div
                className="FormWrapper-hxeBuG qScKG"
              >
                <NoSelectionWrapper>
                  <div
                    className="NoSelectionWrapper-icgMUi jNtnrW"
                  >
                    Select or Create a Quiz
                  </div>
                </NoSelectionWrapper>
              </div>
            </FormWrapper>
          </div>
        </Wrapper>
      </QuizEditor>
    </div>
  </Wrapper>
</QuizEditorPage>
`;

exports[`QuizEditorPage, selection = 1 snapshot 1`] = `
<QuizEditorPage
  clearQuizSelection={[MockFunction]}
  createQuiz={[MockFunction]}
  deleteQuiz={[MockFunction]}
  isNew={false}
  quizzes={
    Array [
      Object {
        "__typename": "Quiz",
        "id": "1",
        "name": "quiz one",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "2",
        "name": "quiz two",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "3",
        "name": "quiz three",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "4",
        "name": "quiz four",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "5",
        "name": "quiz five",
        "type": "pc",
      },
      Object {
        "__typename": "Quiz",
        "id": "6",
        "name": "quiz six",
        "type": "pc",
      },
    ]
  }
  selectQuiz={[MockFunction]}
  selectedQuizId="1"
  updateQuiz={[MockFunction]}
>
  <Wrapper>
    <div
      className="Wrapper-ThVjl hrxOjE"
    >
      <List
        items={
          Array [
            Object {
              "__typename": "Quiz",
              "id": "1",
              "name": "quiz one",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "2",
              "name": "quiz two",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "3",
              "name": "quiz three",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "4",
              "name": "quiz four",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "5",
              "name": "quiz five",
              "type": "pc",
            },
            Object {
              "__typename": "Quiz",
              "id": "6",
              "name": "quiz six",
              "type": "pc",
            },
          ]
        }
        onSelectItem={[MockFunction]}
        selectedItemId="1"
      >
        <Wrapper>
          <div
            className="Wrapper-ThVjl iVApgD"
          >
            <Ul>
              <ul
                className="Ul-bfMUEV cngFat"
              >
                <Li
                  key="1"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={true}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD rxxlO"
                        onClick={[Function]}
                      >
                        quiz one
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="2"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz two
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="3"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz three
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="4"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz four
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="5"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz five
                      </button>
                    </Button>
                  </li>
                </Li>
                <Li
                  key="6"
                >
                  <li
                    className="Li-koaYzM fIJqDv"
                  >
                    <Button
                      isSelected={false}
                      onClick={[Function]}
                    >
                      <button
                        className="Button-kDSBcD eZGGar"
                        onClick={[Function]}
                      >
                        quiz six
                      </button>
                    </Button>
                  </li>
                </Li>
              </ul>
            </Ul>
          </div>
        </Wrapper>
      </List>
      <QuizEditor
        clearQuizSelection={[MockFunction]}
        isNew={false}
        onCreateQuiz={[MockFunction]}
        onDeleteQuiz={[MockFunction]}
        onUpdateQuiz={[MockFunction]}
        quiz={
          Object {
            "__typename": "Quiz",
            "id": "1",
            "name": "quiz one",
            "type": "pc",
          }
        }
      >
        <Wrapper>
          <div
            className="Wrapper-ThVjl eCjdPG"
          >
            <CreateWrapper>
              <div
                className="CreateWrapper-kzyVjd gyhNsT"
              >
                <EditorButton
                  disabled={false}
                  onClick={[Function]}
                >
                  <button
                    className="EditorButton-chDRFm dVoSaf"
                    disabled={false}
                    onClick={[Function]}
                  >
                    New
                  </button>
                </EditorButton>
              </div>
            </CreateWrapper>
            <FormWrapper>
              <div
                className="FormWrapper-hxeBuG qScKG"
              >
                <QuizForm
                  clearQuizSelection={[MockFunction]}
                  isNew={false}
                  onCreateQuiz={[MockFunction]}
                  onDeleteQuiz={[MockFunction]}
                  onUpdateQuiz={[MockFunction]}
                  quiz={
                    Object {
                      "__typename": "Quiz",
                      "id": "1",
                      "name": "quiz one",
                      "type": "pc",
                    }
                  }
                >
                  <Form
                    onSubmit={[Function]}
                  >
                    <form
                      className="Form-dhHGoS fGKlyR"
                      onSubmit={[Function]}
                    >
                      <FieldsWrapper>
                        <div
                          className="FieldsWrapper-eDhoxC fXMMNv"
                        >
                          <Label
                            htmlFor="name"
                          >
                            <label
                              className="Label-bxMjsr hSpJCw"
                              htmlFor="name"
                            >
                              name:
                              <InputText
                                id="name"
                                name="name"
                                onChange={[Function]}
                                placeholder="Enter a name"
                                type="text"
                                value="quiz one"
                              >
                                <input
                                  className="InputText-ektxEQ kaKplR"
                                  id="name"
                                  name="name"
                                  onChange={[Function]}
                                  placeholder="Enter a name"
                                  type="text"
                                  value="quiz one"
                                />
                              </InputText>
                            </label>
                          </Label>
                          <Label
                            htmlFor="type"
                          >
                            <label
                              className="Label-bxMjsr hSpJCw"
                              htmlFor="type"
                            >
                              type:
                              <input
                                checked={true}
                                id="type"
                                name="type"
                                onChange={[Function]}
                                type="radio"
                                value="pc"
                              />
                            </label>
                          </Label>
                        </div>
                      </FieldsWrapper>
                      <ButtonsWrapper>
                        <div
                          className="ButtonsWrapper-xBhhP evhcHL"
                        >
                          <EditorButton
                            disabled={true}
                            type="submit"
                          >
                            <button
                              className="EditorButton-chDRFm dVoSaf"
                              disabled={true}
                              type="submit"
                            >
                              Save
                            </button>
                          </EditorButton>
                          <EditorButton
                            disabled={false}
                            onClick={[Function]}
                          >
                            <button
                              className="EditorButton-chDRFm dVoSaf"
                              disabled={false}
                              onClick={[Function]}
                            >
                              Delete
                            </button>
                          </EditorButton>
                        </div>
                      </ButtonsWrapper>
                    </form>
                  </Form>
                </QuizForm>
              </div>
            </FormWrapper>
          </div>
        </Wrapper>
      </QuizEditor>
    </div>
  </Wrapper>
</QuizEditorPage>
`;
